src/apps/mixed_fem_helper.hpp

```diff
@@ -0,0 +1,942 @@
+    qpts1D[4] = 1.0;
+
+    // Order 5 Gauss-Lobatto Weights
+    wgts1D[0] = 0.05;
+    wgts1D[1] = 0.272222222222222222;
+    wgts1D[2] = 0.355555555555555556;
+    wgts1D[3] = 0.272222222222222222;
+    wgts1D[4] = 0.05;
+
+    break;
+
+  }
+
+}
+
+RAJA_HOST_DEVICE
+inline void LinAlg_qrule_Legendre(
+  int    order,
+  double *qpts1D,
+  double *wgts1D)
+{
+  // Define the Gauss-Legendre quadrature points and weights over the
+  // 1D domain [0,1] for rules up to order 5
+  switch( order ) {
+
+  case 1 :
+
+    // Order 1 Gauss-Legendre Points
+    qpts1D[0] = 0.5;
+
+    // Order 1 Gauss-Legendre Weights
+    wgts1D[0] = 1.0;
+
+    break;
+
+  case 2 :
+
+    // Order 2 Gauss-Legendre Points
+    qpts1D[0] = 0.211324865405187118;
+    qpts1D[1] = 0.788675134594812882;
+
+    // Order 2 Gauss-Legendre Weights
+    wgts1D[0] = 0.5;
+    wgts1D[1] = 0.5;
+
+    break;
+
+  case 3 :
+
+    // Order 3 Gauss-Legendre Points
+    qpts1D[0] = 0.112701665379258311;
+    qpts1D[1] = 0.5;
+    qpts1D[2] = 0.887298334620741689;
+
+    // Order 3 Gauss-Legendre Weights
+    wgts1D[0] = 0.277777777777777778;
+    wgts1D[1] = 0.444444444444444444;
+    wgts1D[2] = 0.277777777777777778;
+
+    break;
+
+  case 4 :
+
+    // Order 4 Gauss-Legendre Points
+    qpts1D[0] = 0.0694318442029737124;
+    qpts1D[1] = 0.330009478207571868;
+    qpts1D[2] = 0.669990521792428132;
+    qpts1D[3] = 0.930568155797026288;
+
+    // Order 4 Gauss-Legendre Weights
+    wgts1D[0] = 0.173927422568726929;
+    wgts1D[1] = 0.326072577431273071;
+    wgts1D[2] = 0.326072577431273071;
+    wgts1D[3] = 0.173927422568726929;
+
+    break;
+
+  case 5 :
+
+    // Order 5 Gauss-Legendre Points
+    qpts1D[0] = 0.0469100770306680036;
+    qpts1D[1] = 0.230765344947158454;
+    qpts1D[2] = 0.5;
+    qpts1D[3] = 0.769234655052841546;
+    qpts1D[4] = 0.953089922969331996;
+
+    // Order 5 Gauss-Legendre Weights
+    wgts1D[0] = 0.118463442528094544;
+    wgts1D[1] = 0.239314335249683234;
+    wgts1D[2] = 0.284444444444444444;
+    wgts1D[3] = 0.239314335249683234;
+    wgts1D[4] = 0.118463442528094544;
+
+    break;
+
+  }
+
+}
+
+RAJA_HOST_DEVICE
+inline void get_quadrature_rule(
+  const int    quad_type,
+  const int    quad_order,
+  double       (&qpts_1d)[MAX_QUAD_ORDER],
+  double       (&wgts_1d)[MAX_QUAD_ORDER])
+{
+  // Generate the 1D set of points and weights over the interval [0,1]
+  switch( quad_type ) {
+
+  case 0 :
+    LinAlg_qrule_Lobatto(quad_order, qpts_1d, wgts_1d);
+    break;
+
+  case 1 :
+    LinAlg_qrule_Legendre(quad_order, qpts_1d, wgts_1d);
+    break;
+
+  }
+}
+
+RAJA_HOST_DEVICE
+constexpr double compute_detj(
+  const double jxx,
+  const double jxy,
+  const double jxz,
+  const double jyx,
+  const double jyy,
+  const double jyz,
+  const double jzx,
+  const double jzy,
+  const double jzz)
+{
+  return
+    jxy*jyz*jzx - jxz*jyy*jzx + jxz*jyx*jzy -
+    jxx*jyz*jzy - jxy*jyx*jzz + jxx*jyy*jzz;
+}
+
+template<int M>
```

Comment:
Would this make more sense as constexpr instead of a non type template parameter?

Reply:
Resolved in person